clusterloader2/pkg/measurement/common/loadbalancer_nodesync_latency.go

```diff
@@ -0,0 +1,283 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/runtimeobjects"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+	"time"
+)
+
+const (
+	loadBalancerNodeSyncLatencyName = "LoadBalancerNodeSyncLatency"
+	defaultNodeSyncLatencyTimeout   = 30 * time.Minute
+
+	// labelNodeExcludeBalancers is the node label to exclude a node from being a LB backend
+	labelNodeExcludeBalancers = "node.kubernetes.io/exclude-from-external-load-balancers"
+	// nodeSyncEventReason is the event reason emitted by service controller when it completes node sync on the lb.
+	nodeSyncEventReason = "UpdatedLoadBalancer"
+
+	phaseNodeSyncStart    = "nodesync_triggered"
+	phaseNodeSyncComplete = "nodesync_complete"
+)
+
+// Scheme is the default instance of runtime.Scheme to which types in the Kubernetes API are already registered.
+// This is needed for ConfigMap search.
+var Scheme = runtime.NewScheme()
```

Comment:
Can we avoid this?

I see this is used only in one place for Search in events.
Can we instead create an appropriate field selector outselves?

It should actually be as simple as:
```
selector := client.CoreV1().Events(s.selector.Namespace).GetFieldSelector(svc.Name, svc.Namespace, nil, nil)
```

Reply:
Good call.